### ISSUE TITLE
Support CTEs in ALTER TABLE DROP PARTITION

### ISSUE THREAD

**Issue by puzpuzpuz** [questdb/questdb#2878]
### Is your feature request related to a problem?

ATM it's not possible to drop the oldest partition or oldest N partitions with a single statements. Would be nice to add support for CTEs in ALTER TABLE statements to let users run statements like the following one:
```sql
with m as (select min(timestamp) min_timestamp from trades)
alter table trades drop partition where timestamp = m.min_timestamp;
```

**Comment by ideoma** [comment]
You mean not possible using single statement?

We do not expose information about the partitions the result queries only get results about the time ranges. You can DROP all partitions older than point in time. Why would you want to drop N oldest partitions if there is no way to know how many partitions you have?

**Comment by puzpuzpuz** [comment]
> You mean not possible using single statement?

Exactly. That's why the issue has "with a single SQL statement" in its name.

> Why would you want to drop N oldest partitions if there is no way to know how many partitions you have?

AFAIR @jsjain wanted to drop the older partition with a single statement. Doing that with two statements is possible, but slightly less convenient.

**Comment by marregui** [comment]
this can be achieved with:

1. figure out the last partition's timestamp:
     ```sql
     select ts from table limit -1;
     ```

2. copy the resulting timestamp, say `2023-12-22T14:06:31.819131Z` and paste like so:
      ```sql
      alter table trade drop partition list '2023-12-22T14:06:31.819131Z';
      ```

`LIST` is the way to go when deleting partitions, as per https://github.com/questdb/questdb/pull/3002 , as it is safer. A mistake in the original command which contains `IN` and it may be a disaster. With `LIST` the process is more involved and this filters the possibility of making a big mistake.

Would it be OK to close the issue, unless some user comes in with the explicit requirement and a use case?

**Comment by puzpuzpuz** [comment]
> Would it be OK to close the issue, unless some user comes in with the explicit requirement and a use case?

This issue originated from a user, but I don't mind closing it.
